### PR TITLE
feat(directive): added support for host actions

### PR DESCRIPTION
**Pull request by vsavkin**
/cc @IgorMinar

### CODE REVIEW

**Comment by tbosch** [comment]
I think the method on the `Renderer` interface should be this:
```
callAction(viewRef:RenderViewRef, elementIndex:number, actionExpression:string)
```

And in the `ElementBinder`, we store a Map from expression string to actual AST that we will use to execute the action. This is then very similar to `setProperty`, where we also pass in the actual property name and store the Setter on the `ElementBinder`. 

Benefits:

- For JS, we could have a renderer that parses on the fly and allow a directive to communicate directly with the renderer (just as you proposed with `setElementProperty`).
- The api stays independent of directives. This comes in handy if we add view actions, as they can go through the same method call and use the same AST storage on the `ElementBinder`.
- We don't need to rely on the directive id (which is not unique right now...)

Sorry, should have thought about this earlier while we paired...

File: gulpfile.js [hunk 1]
```diff
@@ -385,6 +385,14 @@ gulp.task('test.unit.js', ['build.js.dev'], function (neverDone) {
   });
 });
 
+gulp.task('watch.js.dev', ['build.js.dev'], function (neverDone) {
```
Comment (tbosch) [inline gulpfile.js]: Just to double check: Should this be in this commit?

File: gulpfile.js [hunk 2]
```diff
@@ -410,6 +418,7 @@ gulp.task('test.unit.dart', ['build/tree.dart'], function (done) {
   watch('modules/angular2/**', function() {
     runSequence(
       '!build/tree.dart',
+      'build/format.dart',
```
Comment (tbosch) [inline gulpfile.js]: Same here?

File: modules/angular2/src/render/dom/dom_renderer.js [hunk 3]
```diff
@@ -192,6 +192,11 @@ export class DomRenderer extends Renderer {
     view.setElementProperty(elementIndex, propertyName, propertyValue);
   }
 
+  callAction(viewRef:RenderViewRef, elementIndex:number, actionExpression:string, actionArgs:any):void {
```
Comment (tbosch) [inline modules/angular2/src/render/dom/dom_renderer.js]: could you add a test into `dom_renderer_integration_spec.js` for this as well?

File: gulpfile.js (hunk 1 above)
Comment (IgorMinar) [inline gulpfile.js]: @vsavkin what is this doing here and why was it merged? doesn't `serve.js.dev` accomplish the same?

**Pull request merged**